### PR TITLE
[Compression] Add support for torch 2.0

### PR DESCRIPTION
**Pull request by Bonytu**
#### Test Options ####
  - [ ] fast test
  - [ ] full test - HPO
  - [ ] full test - NAS
  - [ ] full test - compression

### Checklist ###
  - [ ] test case
  - [ ] doc

### CODE REVIEW

File: examples/tutorials/pruning_bert_glue.py [hunk 1]
```diff
@@ -159,11 +159,21 @@ def preprocess_function(examples):
...
+else:
+    from torch.optim.lr_scheduler import _LRScheduler
+    SCHEDULER = _LRScheduler
+
 
 def training(model: torch.nn.Module,
              optimizer: torch.optim.Optimizer,
              criterion: Callable[[torch.Tensor, torch.Tensor], torch.Tensor],
-             lr_scheduler: torch.optim.lr_scheduler._LRScheduler = None,
+             lr_scheduler: SCHEDULER = None, # type: ignore
```
Comment (J-shang) [inline examples/tutorials/pruning_bert_glue.py:176]: after modifying the tutorial, re-build doc is needed (`make en`).
and before `# type: ignore` prefer double space.

File: examples/tutorials/pruning_bert_glue.py [hunk 2]
```diff
@@ -159,11 +159,21 @@ def preprocess_function(examples):
...
 
+from nni.common.version import torch_version_is_2
+
+
+if torch_version_is_2():
+    from torch.optim.lr_scheduler import LRScheduler
+    SCHEDULER = LRScheduler
+else:
+    from torch.optim.lr_scheduler import _LRScheduler
+    SCHEDULER = _LRScheduler
```
Comment (J-shang) [inline examples/tutorials/pruning_bert_glue.py:170]: what about put this `if ... else ...` in some where and we just `from xxx import SCHEDULER` here, then we don't need to copy paste it everywhere.

File: examples/tutorials/pruning_bert_glue.py (hunk 1 above)
Comment (J-shang) [reply examples/tutorials/pruning_bert_glue.py:176]: and why need type ignore here?

Comment (Bonytu) [reply examples/tutorials/pruning_bert_glue.py]: good suggestion, I will modify it.

File: nni/contrib/compression/utils/evaluator.py [hunk 3]
```diff
@@ -656,7 +657,7 @@ def get_dummy_input(self) -> Any:
 
 _OPTIMIZERS = Union[Optimizer, List[Optimizer]]
 _TRAINING_STEP = Callable[..., Union[Tensor, Tuple[Tensor], Dict[str, Tensor]]]
-_SCHEDULERS = Union[None, _LRScheduler, List[_LRScheduler]]
+_SCHEDULERS = Union[None, SCHEDULER, List[SCHEDULER]]  # type: ignore
```
Comment (J-shang) [inline nni/contrib/compression/utils/evaluator.py:660]: what's the type error?

**Review by J-shang — APPROVED** [review]

**Review by QuanluZhang — APPROVED** [review]
